src/screens/monitor/components/content/index.tsx

```diff
@@ -19,18 +20,29 @@ export default function MonitorContent() {
     fetch
   } = useContext(MonitorContext);
 
-  useEffect(() => {
-    const interval = setTimeout(async () => {
+  const startTickLongPulling = useCallback(() => {
+    interval.current = setTimeout(async () => {
       await fetch(page, pageSize, sourcesOfInterest);
       if (page === 0) {
         setTime(new Date());
       }
     }, 5000);
+  }, [fetch, page, pageSize, sourcesOfInterest]);
+
+  useEffect(() => {
+   if (interval.current) {
+     clearTimeout(interval.current);
+     startTickLongPulling();
+   }
+  }, [sourcesOfInterest, startTickLongPulling]);
+
+  useEffect(() => {
+    startTickLongPulling();
 
     return () => {
-      clearTimeout(interval);
+      clearTimeout(interval.current);
     }
-  }, [time]);
+  }, [time, startTickLongPulling]);
```

Comment:
This can be extracted into a hook. Either move the interval logic into a hook or use a package.